### DISCUSSION TITLE
The Python SQL minifier you deserve, but not the one you need.

### DISCUSSION THREAD

**Discussion by philippeboyd** [andialbrecht/sqlparse#757] [Show and tell]
Can't believe this library doesn't have an SQL minifier and no issue nor conversations were open regarding this...

So I've been scratching my head over this library as to why can't we easily minify SQL because sometime we want or have to.

I came up with this solution:

```python
import sqlparse
from sqlparse.sql import Comment as SqlComment
from sqlparse.tokens import Comment
from sqlparse.utils import imt


def minify_sql(sql: str) -> str:
    return " ".join(
        token.value
        for parsed in sqlparse.parse(sql)
        for token in parsed.flatten()
        if not token.is_whitespace and not imt(token, t=Comment, i=SqlComment)
    )


# Example usage:
multiline_sql = """
SELECT
    -- comment 1
    column1,


    -- comment 2
    column2,
    REPLACE(column3, '_', ' ') AS replaced_column
FROM
    table1
WHERE
    condition1 = 'value1'
    AND condition2 = 'value2';
    
"""

minified_sql = minify_sql(multiline_sql)
print(minified_sql)
```

Which will result in 
```sql
SELECT column1 , column2 , REPLACE ( '_' , ' ' ) AS replaced_column FROM table1 WHERE condition1 = 'value1' AND condition2 = 'value2' ;
```

Now, granted, there are spaces before commas or semicolons but that's the best I came up with... I don't want to add a search and replace of `' ,'` and `' ;'` because those string could be inside a valid SQL statement/column definition and whatnot.

Let me know what you think.

**Comment by troylepnc** [comment]
' '.join(sql.split()) works for me, need to exclude single line comments
